pkg/signing/signing.go

```diff
@@ -223,6 +367,70 @@ func SifAddSignature(sinfo *image.Sifinfo, signature []byte) error {
 	return nil
 }
 
+func sykeysFetchPubkey(fingerprint string, sykeysAddr string) (openpgp.EntityList, error) {
+	url := "http://" + sykeysAddr + "/pks/lookup?op=get&options=mr&search=0x" + fingerprint
+	log.Println("url :", url)
+	res, err := http.Get(url)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	el, err := openpgp.ReadArmoredKeyRing(res.Body)
+	res.Body.Close()
+	if err != nil {
+		log.Println("error while trying to read armored key ring: ", err)
+		return nil, err
+	}
+	if len(el) == 0 {
+		err = fmt.Errorf("no keys in keyring")
+		log.Println(err)
+		return nil, err
+	}
+	if len(el) > 1 {
+		err = fmt.Errorf("server returned more than one key for unique fingerprint")
+		log.Println(err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+func sykeysPushPubkey(entity *openpgp.Entity, sykeysAddr string) (err error) {
+	u := "http://" + sykeysAddr + "/pks/add"
+	w := bytes.NewBuffer(nil)
+
+	wr, err := armor.Encode(w, openpgp.PublicKeyType, nil)
+	if err != nil {
+		log.Println("armor.Encode failed")
+	}
+
+	err = entity.Serialize(wr)
+	if err != nil {
+		log.Println("can't serialize public key", err)
+		return err
+	}
+	wr.Close()
```

Comment:
Might be better to `defer wr.Close()` above after you're sure `Encode` has succeeded?

Reply:
Right !

Reply:
This cannot be done there, and got me debugging for a minute there. Closing the file at this point flushes the write buffers and allows the sending of a proper buffer to the server. With deferring the close of "wr", the buffer sent was incomplete and caused errors.